pkg/controller/operands/networkAddons_test.go

```diff
@@ -362,6 +363,59 @@ var _ = Describe("CNA Operand", func() {
 			Expect(req.Conditions).To(BeEmpty())
 		})
 
+		type ovsAnnotationParams struct {
+			annotationExists    bool
+			annotationValue     string
+			ovsDeployExpected   bool
+		}
+		table.DescribeTable("when reconciling ovs-cni", func( o ovsAnnotationParams) {
+			hcoOVSConfig := commonTestUtils.NewHco()
+			hcoOVSConfig.Annotations = map[string]string{}
+
+			if o.annotationExists {
+				hcoOVSConfig.Annotations["deployOVS"] = o.annotationValue
+			}
+
+			existingResource := NewNetworkAddons(hcoOVSConfig)
+
+			cl := commonTestUtils.InitClient([]runtime.Object{hco, existingResource})
+			handler := (*genericOperand)(newCnaHandler(cl, commonTestUtils.GetScheme()))
+			res := handler.ensure(req)
+			Expect(res.UpgradeDone).To(BeFalse())
+			Expect(res.Err).To(BeNil())
+
+			foundResource := &networkaddonsv1.NetworkAddonsConfig{}
+			Expect(
+				cl.Get(context.TODO(),
+					types.NamespacedName{Name: existingResource.Name, Namespace: existingResource.Namespace},
+					foundResource),
+			).To(BeNil())
+
+			if o.ovsDeployExpected {
+				Expect(existingResource.Spec.Ovs).ToNot(BeNil(), "Ovs spec should be added")
+			} else {
+				Expect(existingResource.Spec.Ovs).To(BeNil(), "Ovs spec should not be added")
+			}
+			//Expect(req.Conditions).To(BeEmpty())
```

Comment:
@RamLavi @nunnatsa 
I changed these 3 ovs-opt-in tests to table, as advised.
Looks like the conditions are not always empty, and for the negative tests (ovs should not be deployed), I'm getting that HCO conditions are not empty (see below). Is that expected and I can safely delete `Expect(req.Conditions).To(BeEmpty())` ?

```
• Failure [0.001 seconds]
CNA Operand
/home/ocohen/Repositories/fork/hyperconverged-cluster-operator/pkg/controller/operands/networkAddons_test.go:27
  NetworkAddonsConfig
  /home/ocohen/Repositories/fork/hyperconverged-cluster-operator/pkg/controller/operands/networkAddons_test.go:29
    when reconciling ovs-cni
    /home/ocohen/Repositories/fork/hyperconverged-cluster-operator/pkg/controller/operands/networkAddons_test.go:371
      should not have ovs if deployOVS annotation is not set to true [It]
      /home/ocohen/Repositories/fork/hyperconverged-cluster-operator/pkg/controller/operands/networkAddons_test.go:408

      Expected
          <common.HcoConditions | len:3>: {
              "Available": {
                  Type: "Available",
                  Status: "False",
                  Reason: "NetworkAddonsConfigConditions",
                  Message: "NetworkAddonsConfig resource has no conditions",
                  LastHeartbeatTime: {
                      Time: 0001-01-01T00:00:00Z,
                  },
                  LastTransitionTime: {
                      Time: 0001-01-01T00:00:00Z,
                  },
              },
              "Progressing": {
                  Type: "Progressing",
                  Status: "True",
                  Reason: "NetworkAddonsConfigConditions",
                  Message: "NetworkAddonsConfig resource has no conditions",
                  LastHeartbeatTime: {
                      Time: 0001-01-01T00:00:00Z,
                  },
                  LastTransitionTime: {
                      Time: 0001-01-01T00:00:00Z,
                  },
              },
              "Upgradeable": {
                  Type: "Upgradeable",
                  Status: "False",
                  Reason: "NetworkAddonsConfigConditions",
                  Message: "NetworkAddonsConfig resource has no conditions",
                  LastHeartbeatTime: {
                      Time: 0001-01-01T00:00:00Z,
                  },
                  LastTransitionTime: {
                      Time: 0001-01-01T00:00:00Z,
                  },
              },
          }
      to be empty

```

Reply:
It's not so important. We just need to make sure that upgrade is working with and without the annotation